### PR TITLE
feat(globe): edge-tap to skip tracks while listening

### PR DESCRIPTION
**Pull request by taewanu**
## What

Experimental: tapping the globe's left/right edge skips tracks while a song is
playing. Stacks on #162 (reuses the shared `step()` / `findAdjacentPlayable` via a
globe-chart-store seam, so no adjacency logic is duplicated). Opened as a **draft
for on-device validation** before deciding whether to merge.

## Behavior

- Listening (a track is playing): a no-movement tap on the left or right third
  steps prev/next; the center third still selects the nearest country; a drag
  still spins.
- Not listening: every tap selects a country (edge-skip is off).
- A one-time edge hint, reduced-motion-aware.

## Device-test focus (what decides merge)

- 8px tap-vs-spin threshold (`TAP_MAX_PX`): does an edge skip trigger too easily
  while spinning, or a short flick skip unintentionally?
- 33/33/33 thirds: does the center (select) zone feel too narrow on a phone?
- iOS: left-edge skip vs Safari's system back-swipe.

## Verify

- `format` / `lint` / `typecheck` / `test` (516) all green.

> Base is `feat/mini-player-prev-next` so the diff shows only the edge-tap delta;
> it retargets to `main` once #162 merges.

## Summary by CodeRabbit

* **New Features**
  * Added a double-tap skip hint and a skip confirmation animation on the globe.
  * Improved globe tap behavior so edge taps can skip between tracks, while center taps still select normally.
  * Added a new icon used in the skip hint UI.

* **Bug Fixes**
  * Made skip actions reliably trigger the next visual cue, even when repeated in the same direction.
  * Respected reduced-motion settings by simplifying the skip animation for users who prefer less motion.

### CODE REVIEW

**Comment by taewanu** [comment]
✅ Reviewed [`804ef320e399`](https://github.com/taewanu/sounds-abroad/commit/804ef320e399d473863b72ac92f6253a2f64588c)

_Scope: [`115dbb3d48d0..804ef320e399`](https://github.com/taewanu/sounds-abroad/compare/115dbb3d48d024b03bbb586b191305ca9294bbf5...804ef320e399d473863b72ac92f6253a2f64588c)_

**2 findings · 0 open · 2 resolved**

- [`src/components/globe/globe-scene.tsx:72`](https://github.com/taewanu/sounds-abroad/pull/163#discussion_r3510285187) · resolved
- [`src/components/globe/globe-scene.tsx:75`](https://github.com/taewanu/sounds-abroad/pull/163#discussion_r3510345118) · resolved

<details><summary>8 files</summary>

- `src/app/chart-screen.tsx`
- `src/app/globals.css`
- `src/components/chart-sheet/sheet.tsx`
- `src/components/globe/edge-tap-hint.tsx`
- `src/components/globe/globe-scene.tsx`
- `src/components/globe/skip-flash.tsx`
- `src/components/globe/spin-snap-controls.tsx`
- `src/lib/globe-chart-store.test.ts`

</details>

<details><summary>Reviewed 4 commits</summary>

- `527d59ea` · 2026-07-02 03:30 UTC
- `115dbb3d` · 2026-07-02 03:50 UTC
- `72d5865e` · 2026-07-02 04:41 UTC
- `804ef320` · 2026-07-02 05:01 UTC

</details>

🤖 _pr-review-agent_

File: src/components/globe/globe-scene.tsx [hunk 1]
```diff
@@ -61,6 +61,18 @@ function SceneContent() {
...
   const readMode = useGlobeChart((s) => s.readMode);
+  const listening = useGlobeChart((s) => s.listening);
+  const skip = useGlobeChart((s) => s.skip);
+
+  // Route the edge-tap skip and signal the directional cue from one place: the
+  // decision still lives in the controls, this only forwards and flashes.
+  const handleSkip = useCallback(
+    (dir: 1 | -1) => {
+      skip(dir);
+      globeChartStore.getState().signalSkip(dir);
```
Comment (taewanu) [inline src/components/globe/globe-scene.tsx]: _🐛 bug_ | _🟡 nit_

**Gate the skip cue on an actual track change.**

- `findAdjacentPlayable` clamps at the first/last playable track, so `step(dir)` silently no-ops there
- `signalSkip` fires regardless, so the chevron flashes a "skip happened" cue when nothing played
- MiniPlayer already disables prev/next at the boundaries via `canPrev`/`canNext`; the edge-tap path skips that gate

🤖 _pr-review-agent_

**Review by taewanu — COMMENTED** [review]
_pr-review-agent v0.2.3 (preview release). [Report a problem](https://github.com/taewanu/pr-review-agent/issues)._

---

Edge-tap-to-skip is well-built: gesture classification, store wiring, and unit tests all hang together.

- One boundary case flashes the directional cue when `step` no-ops at the first/last playable track
- Nothing else high-signal to flag

---

🤖 _Auto-submitted by [pr-review-agent](https://github.com/taewanu/pr-review-agent). Edit as needed._

File: src/components/globe/globe-scene.tsx (hunk 1 above)
Comment (taewanu) [reply src/components/globe/globe-scene.tsx]: Fixed in `115dbb3`. `step(dir)` now returns whether a track actually changed (false when `findAdjacentPlayable` clamps at the first/last playable track), and the globe's `handleSkip` fires `signalSkip` only on a real change. The edge-tap path now respects the boundary the same way the mini-player's `canPrev`/`canNext` disable the buttons, so the cue no longer claims a skip that didn't happen. Good catch.

Comment (taewanu) [reply src/components/globe/globe-scene.tsx]: _Neither half of the change landed_ [`527d59e:L72`](https://github.com/taewanu/sounds-abroad/blob/527d59ea220eb9ba9f12d9d8e1d45cb5fc63124a/src/components/globe/globe-scene.tsx#L72)

The cue still fires regardless of whether a track actually played.

- `step` has no return value, so there is nothing for the caller to gate on
- `handleSkip` still calls `signalSkip` unconditionally right after `skip`

🤖 _pr-review-agent_

File: src/components/globe/globe-scene.tsx [hunk 2]
```diff
@@ -61,6 +61,21 @@ function SceneContent() {
...
+
+  // Route the edge-tap skip and signal the directional cue from one place: the
+  // decision still lives in the controls; this forwards, then flashes only when
+  // a track actually changed.
+  const handleSkip = useCallback(
+    (dir: 1 | -1) => {
+      // step clamps (returns false) at the first/last playable track, mirroring
+      // the mini-player's disabled prev/next, so the cue never claims a skip
+      // that didn't happen.
+      if (skip(dir)) globeChartStore.getState().signalSkip(dir);
```
Comment (taewanu) [inline src/components/globe/globe-scene.tsx]: _🔧 refactor_ | _🟡 nit_

**Pin the suppress-on-clamp path with a test.**

- The cue-only-on-real-skip contract is the reason this PR exists, and nothing asserts it
- `signalSkip` and `findAdjacentPlayable` are tested in isolation, but not the wiring that gates one on the other
- A regression to `signalSkip(dir)` unconditional would flash on a clamped edge-tap and no test would catch it

🤖 _pr-review-agent_

**Review by taewanu — COMMENTED** [review]
_pr-review-agent v0.2.3 (preview release). [Report a problem](https://github.com/taewanu/pr-review-agent/issues)._

---

Boolean return threaded through `skip`/`step` so the globe cue flashes only on a real track change. Types line up across the store, chart, and scene, and the default no-ops return `false` correctly. One gap: no test pins the new suppress-on-clamp behavior that is the whole point of the change.

---

🤖 _Auto-submitted by [pr-review-agent](https://github.com/taewanu/pr-review-agent). Edit as needed._

File: src/app/chart-screen.tsx [hunk 3]
```diff
@@ -231,6 +248,8 @@ function ChartScreenInner({
         canPrev={canPrev}
         canNext={canNext}
       />
+      <EdgeTapHint active={hasCurrentTrack} />
+      <SkipFlash sheetSnap={snap} />
```
[the remark this answers is not in the document]
Comment (taewanu) [reply src/app/chart-screen.tsx]: Fixed in `804ef32`. `EdgeTapHint`'s `active` is now `hasCurrentTrack && snap !== "full"`, so a track that starts while the sheet covers the globe no longer arms and marks-seen the one-time hint unseen; it shows the first time a track plays with the globe actually visible.

File: src/components/globe/spin-snap-controls.tsx [hunk 4]
```diff
@@ -164,9 +185,11 @@ export function SpinSnapControls({
   // the time this runs — it no-ops, no feedback loop.
   useEffect(() => {
     if (targetCode && targetCode !== sim.current.settledCode) {
+      // An external ?cc= selection cancels any armed edge-tap select.
+      clearPendingSelect();
       settleTo(targetCode);
     }
-  }, [targetCode]);
+  }, [targetCode, clearPendingSelect]);
```
[the remark this answers is not in the document]
Comment (taewanu) [reply src/components/globe/spin-snap-controls.tsx:195]: Fixed in `804ef32`. The external-selection branch now resets `lastEdgeTapAt.current` alongside `clearPendingSelect()`, matching `onCancel` and the skip branch, so a tap soon after an external `?cc=` change can't be misread as a double-tap's second tap.

File: src/components/globe/globe-scene.tsx (hunk 2 above)
Comment (taewanu) [reply src/components/globe/globe-scene.tsx]: Keeping this without a dedicated test. The gate is a two-line inline guard (`if (skip(dir)) signalSkip(dir)`) sitting directly on top of two already-tested units: `findAdjacentPlayable` clamps to `null` at the ends (adjacent-playable.test.ts) and `step` returns that as `false`. Pinning the guard itself needs an R3F pointer-integration harness (mount the canvas, synthesise a timed double-tap at a boundary), whose brittleness outweighs a two-line guard downstream of tested logic. Filing the deeper edge-tap-path coverage under the follow-up issue instead. Resolving.
